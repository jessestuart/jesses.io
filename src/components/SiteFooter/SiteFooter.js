/* @flow */
import { DateTime } from 'luxon'
import { Heart } from 'react-feather'
import React from 'react'

import classNames from 'classnames'

import colors from '../../utils/colors'

enum FooterTheme {
  Light: 'Light',
    Dark: 'Dark',
}

type Props = {
  theme: FooterTheme,
}

const SiteFooter = ({ theme = 'Light' }: Props) => (
  <footer
    className={classNames(
      'b--hot-pink bb bottom-0 bw2 f5-ns f6 flex flex-column fw4 lh-title pa4 w-100',
      {
        'bg-gray-primary moon-gray': theme === SiteFooter.Theme.Dark,
        'bg-near-white black-80': theme === SiteFooter.Theme.Light,
      }
    )}
    style={{
      borderTop:
        theme === SiteFooter.Theme.Light
          ? `1px solid ${colors.secondary.light5}`
          : `3px solid ${colors.secondary.light5}`,
    }}
  >
    <div className="flex justify-center items-center">
      © 2013–{DateTime.local().year} Jesse Stuart
    </div>
    <div className="dib center tc flex-ns justify-center items-center">
      Pixels, bits and bytes hand-crafted with
      <Heart
        className="hot-pink mh1"
        width="15"
        height="15"
        fill={colors.primary.main}
      />
      in NYC.
    </div>
  </footer>
)

// TODO Replace w/ TS enum.
SiteFooter.Theme = {
  Light: 'Light',
  Dark: 'Dark',
}

export default SiteFooter
