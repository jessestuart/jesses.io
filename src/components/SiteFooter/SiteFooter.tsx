import classNames from 'classnames'
import { DateTime } from 'luxon'
import * as React from 'react'
import { Heart } from 'react-feather'

import colors from '../../utils/colors'

export enum FooterTheme {
  Light = 'Light',
  Dark = 'Dark',
}

interface Props {
  theme?: FooterTheme
}

const SiteFooter = ({ theme = FooterTheme.Light }: Props) => (
  <footer
    className={classNames(
      'b--hot-pink bb bottom-0 bw2 f5-ns f6 flex flex-column fw4 lh-title pa4 w-100',
      {
        'bg-gray-primary moon-gray': theme === FooterTheme.Dark,
        'bg-near-white black-80': theme === FooterTheme.Light,
      },
    )}
    style={{
      borderTop:
        theme === FooterTheme.Light
          ? `1px solid ${colors.secondary.light5}`
          : `3px solid ${colors.secondary.light5}`,
    }}
  >
    <div className="flex justify-center items-center">
      © 2013–{DateTime.local().year} Jesse Stuart
    </div>
    <div className="dib center tc flex-ns justify-center items-center">
      Pixels, bits and bytes hand-crafted with
      <Heart
        className="hot-pink mh1"
        width="15"
        height="15"
        fill={colors.primary.main}
      />
      in NYC.
    </div>
  </footer>
)

// TODO Replace w/ TS enum.
// @Deprecated
SiteFooter.Theme = {
  Light: 'Light',
  Dark: 'Dark',
}

export default SiteFooter
