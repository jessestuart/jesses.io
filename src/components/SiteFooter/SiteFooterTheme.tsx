const SiteFooterTheme = {
  Light: 'Light',
  Dark: 'Dark',
}

module.exports = SiteFooterTheme
