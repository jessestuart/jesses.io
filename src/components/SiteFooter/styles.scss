// @import '../../layouts/base.scss';
@import '../../styles/js-tachyons.scss';

.footer-section {
  border-top: 1px solid $gray-600;
  bottom: 0;
  display: flex;
  flex-direction: column;
  font-size: 1.2rem;
  height: 4rem;
  justify-content: center;
  line-height: 1.6rem;
  position: relative;
  text-align: center;
  width: 100%;
}
