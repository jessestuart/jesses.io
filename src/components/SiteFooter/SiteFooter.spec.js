import React from 'react'
import SiteFooter from './SiteFooter'
import renderer from 'react-test-renderer'

test('Render SiteFooter component.', () => {
  const tree = renderer.create(<SiteFooter />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Render SiteFooter component with dark theme.', () => {
  const tree = renderer
    .create(<SiteFooter theme={SiteFooter.Theme.Dark} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
