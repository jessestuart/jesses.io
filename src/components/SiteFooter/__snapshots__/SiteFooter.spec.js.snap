// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SiteFooter component with dark theme. 1`] = `
<footer
  className="b--hot-pink bb bottom-0 bw2 f5-ns f6 flex flex-column fw4 lh-title pa4 w-100 bg-gray-primary moon-gray"
  style={
    Object {
      "borderTop": "3px solid #a58fcd",
    }
  }
>
  <div
    className="flex justify-center items-center"
  >
    © 2013–
    2018
     Jesse Stuart
  </div>
  <div
    className="dib center tc flex-ns justify-center items-center"
  >
    Pixels, bits and bytes hand-crafted with
    <svg
      className="hot-pink mh1"
      fill="#ff3a5c"
      height="15"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
      />
    </svg>
    in NYC.
  </div>
</footer>
`;

exports[`Render SiteFooter component. 1`] = `
<footer
  className="b--hot-pink bb bottom-0 bw2 f5-ns f6 flex flex-column fw4 lh-title pa4 w-100 bg-light-gray"
  style={
    Object {
      "borderTop": "1px solid #a58fcd",
    }
  }
>
  <div
    className="flex justify-center items-center"
  >
    © 2013–
    2018
     Jesse Stuart
  </div>
  <div
    className="dib center tc flex-ns justify-center items-center"
  >
    Pixels, bits and bytes hand-crafted with
    <svg
      className="hot-pink mh1"
      fill="#ff3a5c"
      height="15"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
      />
    </svg>
    in NYC.
  </div>
</footer>
`;
